File: backend/aitrade/api/status.py
"""
System status API — GET /api/status.
"""

from fastapi import APIRouter

from ..config import ALPHA_LAB_PATH, TUSHARE_TOKEN
from ..datasource import datasource_manager

router = APIRouter(prefix="/api", tags=["System"])


@router.get("/status")
async def get_status() -> dict:
    """
    Return system-wide status for health checks and UI readiness.

    Returns:
        dict: System status
            - version: API server version
            - torch_available: PyTorch installed
            - torch_device: cuda/cpu
            - data_path: AlphaLab path
            - tushare_token_set: Whether TUSHARE_TOKEN is configured
            - providers: Registered data source providers
    """
    torch_available = False
    torch_device = "N/A"
    try:
        import torch
        torch_available = True
        torch_device = "cuda" if torch.cuda.is_available() else "cpu"
    except ImportError:
        pass

    providers_info = []
    for info in datasource_manager.get_all_providers_info():
        providers_info.append({
            "name": info.name,
            "priority": info.priority,
            "status": info.status,
            "description": info.description,
        })

    return {
        "version": "1.0.0",
        "torch_available": torch_available,
        "torch_device": torch_device,
        "data_path": str(ALPHA_LAB_PATH),
        "tushare_token_set": bool(TUSHARE_TOKEN),
        "providers": providers_info,
    }

File: frontend/src/api/status.ts
/**
 * 系统状态 API 服务对象——封装与后端 `/api/status` 端点的 HTTP 交互。
 *
 * 用于仪表板等场景，读取后端整体可用性（torch 是否安装、数据路径、数据提供方列表等）。
 */
import api from './client'
import type { SystemStatus } from '../types/common'

export const statusService = {
  /**
   * 获取后端系统状态快照。
   *
   * @returns {@link SystemStatus}，包含版本号、torch 可用性、数据路径及数据提供方列表。
   */
  getStatus: () =>
    api.get<SystemStatus>('/api/status').then((r) => r.data),
}
